brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/accountverification/AccountVerificationViewModel.kt

```diff
@@ -144,4 +174,15 @@ class AccountVerificationViewModel(
             )
         }
     }
+
+    private fun loadProfile() {
+        viewModelScope.launch(Dispatchers.IO) {
+            val response = registrationApi.getProfile()
+            val event = when (response.status) {
+                Status.SUCCESS -> AccountVerificationContract.Event.ProfileLoaded(response.data!!)
```

Comment:
Are we sure that response.data and response.message are always non-null? If they are it will lead to NPE. Optionally we could also check that those are not null and add else statement with generic error like this:
`when {
  response.status == Status.SUCCESS && response.data != null -> ...
  response.status == Status.ERROR && response.message != null -> ...
  else -> AccountVerificationContract.Event.ProfileLoadFailed(getString(R.string.generic_error))
}`

Reply:
Yeah that was a bug attractive code that I've written, so I've implemented your suggestion.